pandas/tests/io/parser/test_parsers.py

```diff
@@ -99,3 +102,14 @@ def read_table(self, *args, **kwds):
         kwds = kwds.copy()
         kwds['engine'] = self.engine
         return read_table(*args, **kwds)
+
+
+class TestParameterValidation(object):
```

Comment:
Also, why are we writing a round trip test?  This can be much simpler:
~~~python
data = "a\n1"
msg = "Unknown engine"
with tm.assert_raises_regex(ValueError, msg):
  read_csv(StringIO(data), engine='pyt')
~~~
Oh and yes, use `tm.assert_raises_regex` instead of the `pytest.raises(...)` (`pandas` regex matching is a little more compact).